app/contracts/custom_fields/hierarchy/update_item_contract.rb

```diff
@@ -52,6 +52,7 @@ class UpdateItemContract < Dry::Validation::Contract
 
       rule(:short) do
         next if schema_error?(:item)
+        next unless key?
```

Comment:
Semantical question (note that I am new to `Dry::Validation::Contract`): I understood that our problem was that the value `""` (empty string) could collide in the uniqueness check, though those should be ignored. I.e. two items with an empty short should not trigger a validation error.

However, the rule as specified now, doesn't care about `value` (e.g. `next if value.blank?`) but instead it checks the presence of `key`.

I assume this means something like "if an argument named `:short` was given, run this rule, otherwise skip it".

Is this what we want?

In other words: Would this implementation still work properly, if we defined `starkiller` in our tests as

```
# short is mentioned explicitly as an empty string
let!(:starkiller) { create(:hierarchy_item, label: "starkiller", short: "", parent: vader) }
```

(I might be missing a lot of relevant context here, but I'll post the question at least for the sake of my own education)

Reply:
Yep, in the persistence layer we actually filter out empty strings and such, so that on the contract side, the hash won't contain the key `short` anymore.